algorithms/unit_tests/TestStdAlgorithmsTeamForEach.cpp

```diff
@@ -0,0 +1,157 @@
+//
+// Questions? Contact Christian R. Trott crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <TestStdAlgorithmsCommon.hpp>
+
+namespace Test {
+namespace stdalgos {
+namespace TestForEach {
+
+namespace KE = Kokkos::Experimental;
+
+template <class ValueType>
+struct PrefixIncrementFunctor {
+  KOKKOS_INLINE_FUNCTION
+  void operator()(ValueType& val) const { ++val; }
+};
+
+template <class ValueType>
+struct PlusTwoIncrementFunctor {
+  KOKKOS_INLINE_FUNCTION
+  void operator()(ValueType& val) const { val += 2; }
+};
+
+template <class DataViewType, class UnaryPredType>
+struct TestFunctorA {
+  DataViewType m_dataView;
+  int m_apiPick;
+  UnaryPredType m_unaryPred;
+
+  TestFunctorA(const DataViewType dataView, int apiPick,
+               UnaryPredType unaryPred)
+      : m_dataView(dataView), m_apiPick(apiPick), m_unaryPred(unaryPred) {}
+
+  template <class MemberType>
+  KOKKOS_INLINE_FUNCTION void operator()(const MemberType& member) const {
+    const auto myRowIndex = member.league_rank();
+    auto myRowViewFrom = Kokkos::subview(m_dataView, myRowIndex, Kokkos::ALL());
+
+    switch (m_apiPick) {
+      case 0: {
+        KE::for_each(member, KE::begin(myRowViewFrom), KE::end(myRowViewFrom),
+                     m_unaryPred);
+        break;
+      }
+
+      case 1: {
+        KE::for_each(member, myRowViewFrom, m_unaryPred);
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(std::size_t numTeams, std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level for_each
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  constexpr auto lowerBound = ValueType{5};
+  constexpr auto upperBound = ValueType{523};
+  Kokkos::pair bounds{lowerBound, upperBound};
+  auto [dataView, dataView_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols, bounds, "dataView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  PrefixIncrementFunctor<ValueType> unaryPred;
+
+  // use CTAD for functor
+  TestFunctorA fnc(dataView, apiId, unaryPred);
+  Kokkos::parallel_for(policy, fnc);
+
+  // -----------------------------------------------
+  // run cpp-std kernel and check
+  // -----------------------------------------------
+  for (std::size_t i = 0; i < dataView_h.extent(0); ++i) {
+    auto rowFrom = Kokkos::subview(dataView_h, i, Kokkos::ALL());
```

Comment:
you cannot run the std algo on dataView_h because on host dataView_h and dataView refer to the same allocation. 
Whenever an algorithm changes the data of a view, you need to first make a new view, copy data into it, and then run the std algorithm on that. Otherwise on host you basically only see the std effect.

Reply:
also, for such test, since it is doing something trivial, we don't necessarily have to call the std algorithm. We can just verify the data manually since it is just incremented by 1 wrt origianl data. Look at how the test for `fill` is done

Reply:
Done.